### PR TITLE
chore: add Apache License 2.0

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/go-square/issues/26

### CODE REVIEW

**Comment by rootulp** [comment]
Tagging @Karotoka for review. Sorry I can't add you as a reviewer for some reason. Perhaps we need to add you to celestiaorg?

**Review by evan-forbes — DISMISSED** [review]

**Comment by Karotoka** [comment]
License import approved.
Please delete "and onwards" (I can submit the minor change if its easier).

File: NOTICE [hunk 1]
```diff
@@ -0,0 +1,2 @@
+go-square
+Copyright 2024 and onwards Strange Loop Labs AG
```
Comment (rootulp) [inline NOTICE:2]:
```suggestion
Copyright 2024 Strange Loop Labs AG
```

**Comment by rootulp** [comment]
Done, thanks for the heads up @Karotoka

**Review by cmwaters — APPROVED** [review]
utAck

**Review by distractedm1nd — APPROVED** [review]

**Pull request merged**